test/over_react_redux/value_mutation_checker_test.dart

```diff
@@ -0,0 +1,154 @@
+          final map = {'a': 1, 'b': '2'};
+          final domProps = DomProps(react.a);
+          expect(hasher.hash(map), 2);
+          expect(hasher.hash(domProps.props), 0);
+
+          map['c'] = 3;
+          domProps.addProps({'href': 'google.com'});
+          expect(hasher.hash(map), 3);
+          expect(hasher.hash(domProps.props), 1);
+        });
+
+        test('returns the correct value for an iterable', () {
+          final list = [1, 2, 3];
+          final set = {1, 2, 3};
+          expect(hasher.hash(list), 3);
+          expect(hasher.hash(set), 3);
+
+          list.add(4);
+          set.add(4);
+          expect(hasher.hash(list), 4);
+          expect(hasher.hash(set), 4);
+        });
+      });
+    });
+
+    group('CollectionShallowHasher', () {
+      sharedHashTests(useLengthHasher: false);
+
+      group('hash', () {
+        CollectionShallowHasher hasher;
+
+        setUp(() {
+          hasher = CollectionShallowHasher();
+        });
+
+        test('is not affected by order for sets and maps', () {
+          final set1 = {1, 2, 3};
+          final set2 = {2, 3, 1};
+
+          final map1 = {'a': 1, 'b': 2, 'c': 3};
+          final map2 = {'b': 2, 'a': 1, 'c': 3};
+
+          expect(hasher.hash(set1), hasher.hash(set2));
+          expect(hasher.hash(map1), hasher.hash(map2));
+        });
+
+        test('is affected by order for iterables (besides sets)', () {
+          final list1 = [1, 2, 3];
+          final list2 = [2, 1, 3];
+
+          expect(hasher.hash(list1) == hasher.hash(list2), isFalse);
+        });
+      });
+    });
+  });
+}
+
+void sharedHashTests({bool useLengthHasher}) {
+  InstanceHasher hasher;
+
+  setUp(() {
+    hasher =
+        useLengthHasher ? CollectionLengthHasher : CollectionShallowHasher();
+  });
+
+  group('canHash returns', () {
+    test('false if the object is not a map or an iterable', () {
+      expect(hasher.canHash(0), isFalse);
+      expect(hasher.canHash('test'), isFalse);
+      expect(hasher.canHash(false), isFalse);
+    });
+
+    test('true if the object is a map or iterable', () {
+      final props = DomProps(react.a);
+      final list = MockList();
+
+      expect(hasher.canHash({'e', 'a', 'b'}), isTrue);
+      expect(hasher.canHash([]), isTrue);
+      expect(hasher.canHash({'a': 1}), isTrue);
+      expect(hasher.canHash(props), isTrue);
+      expect(hasher.canHash(list), isTrue);
+    });
+  });
+
+  test('hash throws an argument error if the value is not an iterable or a map',
+      () {
+    expect(() => hasher.hash(0), throwsArgumentError);
+  });
+
+  group('hashHasChanged', () {
+    group('returns false if', () {
+      test('canHash returns false', () {
+        expect(hasher.hasHashChanged(0), isFalse);
+      });
+
+      test('the object has not been checked before', () {
+        final props = DomProps(react.a);
+        expect(hasher.hasHashChanged(props), isFalse);
+      });
+
+      test('the object has changed', () {
```

Comment:
This test is misleading since the object hasn't changed, but rather the variable has changed. I'm not sure we need this test.

We should, however, have a test that verifies it returns false when the object does not change